guacamole/src/main/webapp/app/index/config/indexRouteConfig.js

```diff
@@ -88,7 +88,7 @@ angular.module('index').config(['$routeProvider', '$locationProvider',
 
                 // Otherwise, reject and reroute
                 else {
-                    $location.url(homePage.url);
+                    $location.path(homePage.url);
```

Comment:
Yep. Lacking this change, URL parameters are wiped out when the user is routed to their only connection.

Without the share key continuing to be present in the URL, reloading the tab fails results in a login page rather than the connection they were just using (a pretty severe violation of the [Principle of Least Surprise](https://en.wikipedia.org/wiki/Principle_of_least_astonishment)). Navigating back to the connection after visiting a different page would fail for the same reason.